sqld/src/query_analysis.rs

```diff
@@ -50,6 +50,7 @@ impl StmtKind {
                 | Stmt::CreateIndex { .. },
             ) => Some(Self::Write),
             Cmd::Stmt(Stmt::Select { .. }) => Some(Self::Read),
+            Cmd::Stmt(Stmt::Pragma { .. }) => Some(Self::Other),
```

Comment:
Is this safe? As far as I remember HTTP requests are load balanced over a pool of Libsql connections. Isn't it possible for a client to get their pragma statement executed on one of the connections of the pool and the next of their requests being executed on another connection in the pool (which does not have the pragma)? @psarna. CC @MarinPostma

Reply:
It's safe only for readonly pragmas that don't change state (lots of them are like that), we should refuse ones that change the connection state.

Reply:
Wouldn't it be a bit safer to just allow PRAGMA for Hrana and then give them the category of `Write`?

Reply:
We could just whitelist a few readonly statements and add items to the list if somebody reports an issue

Reply:
I guess whitelisting makes more sense. The question is whether pragma's should be executed always on primary or for some of them it's ok to be run on the replica?

Reply:
there are two things to consider in my opinion: 
i) pragmas that are readonly and generally safe, should be whitelisted
ii) pragmas that change the state of a connection, which cause 2 problems: 1) how to make sure the replica and master share the same connection state, and 2) in what context should such pragmas be allowed (ok for hrana, and not ok for HTTP for instance)